test/test_foreach.py

```diff
@@ -911,8 +911,58 @@ def test_outplace_with_invalid_grads(self, device, dtype, op):
         self.assertIsNotNone(sample.input[0].grad)
         self.assertIsNone(sample.input[1].grad)
 
+    @ops(
+        foreach_unary_op_db + foreach_binary_op_db + foreach_pointwise_op_db + foreach_lerp_op_db,
+        dtypes=OpDTypes.supported,
+        allowed_dtypes=(torch.float64, torch.complex128),
+    )
+    def test_inplace_forward_mode_AD(self, device, dtype, op):
+        if not op.supports_forward_ad:
+            self.skipTest("forward AD not supported")
+
+        # note(crcrpar): These patterns are to skip "sample"s that are incompatible with some
+        # in-place foreach functions' forward path, not forward mode AD.
+        # Raising / xfailing them would rather lose coverage since `sample_inputs` yields
+        # different Scalars and ScalarLists.
+        patterns_to_tolerate = "|".join([
+            _BOOL_SUB_ERR_MSG,
+            # Some non-unary in-place ops with scalar/scalarlist would error out if there's non float ones.
+            "can't be cast to the desired",
+            "clamp is not supported for complex types",
+            "Found dtype Double but expected ComplexDouble",
+        ])
+
+        # this pattern goes to the slow path implementation, needs to update the op db?
+        if dtype == torch.complex128 and op.inplace_variant == torch._foreach_abs_:
+            self.skipTest("In-place abs is not supported for complex tensors.")
```

Comment:
Could we set this in the OpInfo instead (and make it a xfail instead if possible)?